wdl/downsampling_pipeline_part1_dynamic.wdl

```diff
@@ -0,0 +1,154 @@
+version 1.0
+
+workflow downSampling_01 {
+
+    #################################################################################
+    ####        Required basic arguments for downsampling part 1                    #
+    #################################################################################
+    
+    input {
+        File bam_or_cram_file
+        File reference_fasta
+        Int? cram_to_bam_disk_size
+        String? cram_to_bam_mem_size
+        String downsample_docker
+
+        Boolean is_bam_ = basename(bam_or_cram_file, ".bam") + ".bam" == basename(bam_or_cram_file)
+
+        Int bam_to_fq_disk_size
+        String bam_to_fq_mem_size    
+
+    }
+
+    parameter_meta {
+      bam_or_cram_file: ".bam or .cram file to search for SVs. bams are preferable, crams will be converted to bams."
+      bam_file: ".bam file to search for SVs. bams are preferable, crams will be converted to bams."
+      reference_fasta: ".fasta file with reference used to align bam or cram file"
+    }
+
+    meta {
+        author: "Stephanie Hao"
+        email: "shao@broadinstitute.org"
+    }
+    
+    #################################################################################
+    ####        Convert cram to bam                                                 #
+    #################################################################################
+    if (!is_bam_) {
+        call cramToBam {
+            input :
+            cram_file = bam_or_cram_file,
+            reference_fasta = reference_fasta,
+            downsample_docker = downsample_docker,
+            disk_size = cram_to_bam_disk_size,
+            mem_size = cram_to_bam_mem_size
+        }
+    }
+
+
+    #################################################################################
+    ####        Convert bam to fq1 & fq2                                            #
+    #################################################################################
+    call bamToFq { 
+        input :
+            bam_file = select_first([cramToBam.bam_file, bam_or_cram_file]),
+            downsample_docker = downsample_docker,
+            disk_size = bam_to_fq_disk_size,
+            mem_size = bam_to_fq_mem_size
+    }
+
+
+
+    output {
+        File? outputBam = cramToBam.bam_file
```

Comment:
It doesn’t look like the outputBam is used at all in 02, so should it not be a top-level workflow output? (By that I mean it would be a task output but not a workflow output. This would make it easier to clean up intermediate files.)

Or, if it is necessary to output it, I would consider making it a required output - if it's used in a later workflow this will make it easier, and you can just have it copy over the input if it was already a bam